### PR TITLE
FEATURE: EGL-292 - show date filter visibility icon

### PR DESCRIPTION
**Pull request by khanhhuynh-gooddata**
JIRA: EGL-292

---

Supported PR commands:

| Command                                                 | Description                                                |
| ------------------------------------------------------- | ---------------------------------------------------------- |
| `ok to test`                                            | Re-run standard checks                                     |
| `extended check sonar`                                  | SonarQube tests                                            |
| `extended test - backstop`                              | BackstopJS tests                                           |
| **E2E Cypress tests commands - TIGER**                  |                                                            |
| `extended test - tiger-cypress - isolated <testName>`   | Run isolated tests running against recorded Tiger backend. |
| `extended test - tiger-cypress - record <testName>`     | Create a new recording for isolated Tiger tests.           |
| `extended test - tiger-cypress - integrated <testName>` | Run integrated tests against live backend                  |
| **E2E Cypress tests commands - BEAR**                   |                                                            |
| `extended test - cypress - isolated <testName>`         | Run isolated tests running against recorded Bear backend.  |
| `extended test - cypress - record <testName>`           | Create a new recording for isolated Bear tests.            |
| `extended test - cypress - integrated <testName>`       | Run integrated tests against live backend                  |
| **Compatibility matrix test commands - TIGER Backend**  |                                                            |
| `extended test - matrix-test <AIO_version>`             | Run integrated tests against AIO versions.                 |

`<testName>` in cypress commands is used to filter specfiles. Example, to run record with BEAR backend

-   Against `dashboard.spec.ts` and `drilling.spec.ts`, execute command `extended test - cypress - record dashboard,drilling`
-   Against all specfiles, execute command `extended test - cypress - record` or `extended test - cypress - record *`

`<AIO_version>` in commands is used to start test with multiple AIO instances - each instance in triggered by one jenkins build

-   To run with `master` and `stable`, execute command `extended test - matrix-test master,stable` or `extended test - matrix-test latest`
-   To run with specific version,ex: `2.3.0` and `2.3.1`, execute command `extended test - matrix-test 2.3.0,2.3.1`
-   In case `<AIO_version>` is empty, read versions from file `compTigerVersions.txt` of this repo

---

# PR Checklist

-   [ ] commit messages adhere to the [commit message guidelines](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#what-should-the-commits-look-like)
-   [ ] review was done by a Code owner [if necessary](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-tell-if-my-pull-request-needs-approval-by-a-code-owner) (if you think it is not necessary, explain the reasoning in the description or in a comment)
-   [ ] `check` passes
-   [ ] `extended test - backstop` passes
-   [ ] `extended test - tiger-cypress - record` to record new mapping files (Tiger BE)
-   [ ] `extended test - cypress - record` to record new mapping files (Bear BE)
-   [ ] `extended test - tiger-cypress - isolated` passes
-   [ ] `extended test - cypress - isolated` passes
-   [ ] `extended test - tiger-cypress - integrated` passes
-   [ ] `rush change` [was run if applicable](https://github.com/gooddata/gooddata-ui-sdk/blob/master/docs/contributing.md#how-do-i-describe-my-changes-for-the-changelog)

### CODE REVIEW

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16127/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17626/ : FAILURE in 8m 47s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16133/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17631/ : FAILURE in 12m 29s

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16135/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17633/ : FAILURE in 12m 34s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16137/ : SUCCESS in 4s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17635/ : SUCCESS in 34m 39s

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16157/ : SUCCESS in 6s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17655/ : SUCCESS in 12m 26s

File: libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+        it("should return icon and tooltip when mode is hidden", () => {
+            const result = getVisibilityIcon("hidden", true, true, mockIntl);
+            expect(result).toMatchSnapshot();
+        });
+
+        it("should return icon and tooltip when mode is readonly", () => {
+            const result = getVisibilityIcon("readonly", true, true, mockIntl);
+            expect(result).toMatchSnapshot();
+        });
```
[the remark this answers is not in the document]
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-dashboard/src/presentation/filterBar/tests/utils.test.ts:32]: Added!
Thank you!

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16158/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17656/ : SUCCESS in 12m 25s

File: libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx [hunk 2]
```diff
@@ -33,19 +29,24 @@ export interface IDateFilterStatePropsIntersection {
...
     dateFormat?: string;
     locale?: string;
     isTimeForAbsoluteRangeEnabled?: boolean;
     weekStart?: WeekStart;
+    /**
+     * Represents a custom icon with associated tooltip information.
+     *
+     * @alpha
+     */
+    customIcon: IFilterButtonCustomIcon;
```
Comment (phong-nguyen-duy) [inline libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx:49]: It should be optional

**Comment by gdgate** [comment]
Build failed (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16159/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17657/ : FAILURE in 2m 49s

File: libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx (hunk 2 above)
Comment (khanhhuynh-gooddata) [reply libs/sdk-ui-filters/src/DateFilter/DateFilter.tsx]: done!

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-oss-check/16161/ : SUCCESS in 3s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17659/ : SUCCESS in 13m 07s

**Review by phong-nguyen-duy — APPROVED** [review]

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ui-sdk-gate-PR4194

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-sonarqube-scan-zuul-gate/3311/ : SUCCESS in 10m 00s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-unit-tests/17672/ : SUCCESS in 12m 20s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-bear-isolated-zuul-pipeline/922/ : SUCCESS in 11m 30s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-merge-compliance-check/4473/ : SUCCESS in 12s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ui-sdk-fossa-scanning-zuul/1191/ : FAILURE in 1m 22s (non-voting)
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-cypress-tiger-isolated-zuul-pipeline/827/ : SUCCESS in 10m 34s
- https://checklist.intgdc.com/job/gooddata-ui-sdk/job/gooddata-ui-sdk-compatibility-pipeline/480/ : SUCCESS in 6m 45s

**Pull request merged**